vertx-web/src/main/java/io/vertx/ext/web/RoutingContext.java

```diff
@@ -180,6 +180,17 @@
    */
   Route currentRoute();
 
+  /**
+   * @return the current router this context is being routed through.
```

Comment:
I think we need to add more detail in the javadoc. Explain that the `Router` instance may be null or not constant during routing.